[Blog Post] How to start a workgroup within the Julia community (#2138)
JuliaLang/www.julialang.org: blog/2025/02/starting-workgroups.md

+++

> title = "So, You Want to Start a Julia Workgroup?"

Reviewer: ```diff
- title = "So, You Want to Start a Julia Workgroup?"
+ title = "So, you want to start a Julia Workgroup?"
```

> title = "So, You Want to Start a Julia Workgroup?"

Reviewer: ```suggestion
title = "So, you want to start a Julia Workgroup?"
```

authors = "Hetarth Shah, George Datseris, Jacob S. Zelko"
published = "10 February 2025"
rss_pubdate = Date(2025, 2, 10)
rss = """Thoughts and advice on starting your own Julia community workgroup."""
+++

## Introduction

Have you ever wondered how to start a workgroup within Julia?
This post is dedicated to you! 

> Read on to learn how to get started and even if you already are in a workgroup, there might be some helpful tips and advice for you here too!

Reviewer: ### Before

```md
Have you ever wondered how to start a workgroup within Julia?
This post is dedicated to you! 
Read on to learn how to get started and even if you already are in a workgroup, there might be some helpful tips and advice for you here too!
```

### After

```md
Have you ever wondered how to start a workgroup within Julia?

Then, this post is dedicated to you! 
Read on to learn how to get started and even if you already are in a workgroup, there might be some helpful tips and advice for you here too!
```

> Have you ever wondered how to start a workgroup within Julia?
> This post is dedicated to you! 
> Read on to learn how to get started and even if you already are in a workgroup, there might be some helpful tips and advice for you here too!

Reviewer: ```suggestion
Have you ever wondered how to start a workgroup within Julia?

This post is dedicated to you! 
Read on to learn how to get started and even if you already are in a workgroup, there might be some helpful tips and advice for you here too!
```

Not only this but most working groups regularly hold meetings, everyone is free to attend these meetings and discuss about the topics which they want to. Most meeting schedules are posted on Julia's community events calendar but one can also find relevant links in the respective Julia community channels.

## Starting a Group

> How do groups get started within the Julia community?

Starting a workgroup within the Julia community is rather straightforward.
Generally, how most workgroups get started is if there is a particular community (say, the Machine Learning or Documentation ecosystems) that doesn't yet have an active workgroup.
One can post about wanting to start a group for that community in that community's respective platform (whether on Slack, Zulip, etc.) to gauge interest.
If you feel there is enough community interest, go for starting the group and see how it goes!

### Gaining Members

> You created a working group; now what?

From there, it's up to you to set-up the workgroup meeting.

> You can use the [Julia Discourse](discourse.julialang.org/), [Slack](https://julialang.org/slack/#the_julia_language_slack), [Zulip](https://julialang.zulipchat.com), or other communication platforms to announce the creation of a workgroup and also say where workgroup members can find members of this community (usually on Slack or Zulip).

Reviewer: ```suggestion
You can use the [Julia Discourse](discourse.julialang.org/), [Slack](https://julialang.org/slack/#the_julia_language_slack), [Zulip](https://julialang.zulipchat.com), or other communication platforms to announce the creation of a workgroup and also say where workgroup members can find other members of this community (usually on Slack or Zulip).
```

> You can use the [Julia Discourse](discourse.julialang.org/), [Slack](https://julialang.org/slack/#the_julia_language_slack), [Zulip](https://julialang.zulipchat.com), or other communication platforms to announce the creation of a workgroup and also say where workgroup members can find members of this community (usually on Slack or Zulip).
> Then, you can use these platforms to advertise your group -- some folks use a single Discourse thread to announce regular meetings while others use a unique thread per meeting, you could use the respective Slack or Zulip channels to announce your group (as well as the #community channel in Slack) and still others advertise their group across social media. 

Reviewer: ```suggestion
You can use the [Julia Discourse](discourse.julialang.org/), [Slack](https://julialang.org/slack/#the_julia_language_slack), [Zulip](https://julialang.zulipchat.com), or other communication platforms to announce the creation of a workgroup and also say where workgroup members can find members of this community (usually on Slack or Zulip).

Then, you can use these platforms to advertise your group -- some folks use a single Discourse thread to announce regular meetings while others use a unique thread per meeting, you could use the respective Slack or Zulip channels to announce your group (as well as the #community channel in Slack) and still others advertise their group across social media. 
```

You can also post about your working group in other channels like `#general`, `#introduction`, etc. but please be mindful to not post too much as it may come across spammy within the community.

> Now have a working group with enough members in it! Let's move forward with topics: these help in maintaining a healthy environment for your working group.

Reviewer: ```suggestion
Now we have a working group with enough members in it! Let's move forward with topics which help in maintaining a healthy environment for your working group.
```

## Developing an agenda

> What to talk about? 

After creating a workgroup, it makes sense to have a goal, things to work on or discuss with group members. 
What may be helpful is to create an agenda for the workgroup to act as a roadmap for the meeting.
It helps to direct effort and accomplish things within the community faster if you have agenda/roadmap planned out. 
Additionally, loop in your community members to see if they prefer certain activities more than others. 
Take into account all opinions if possible as this will help in preparing your agenda every meeting to have new events and activities. 

> Having a well-planned agenda can also bring other communities. 

Reviewer: ```suggestion
Having a well-planned agenda can also bring in other communities. 
```

> For example, if you plan events that may be interested to a more general Julia audience, you could post in relevant channels about those particular events to see if folks wants to join your community discussion. 

Reviewer: ```suggestion
For example, if you plan events that may be of interest to a more general Julia audience, you could post it in relevant channels about those particular events to see if folks want to join in on your community discussion. 
```

Finally, to help keep folks in your community engaged, consider introducing some fun agenda topics such as regular shout-outs to new talent joining or appreciating major contributors each meeting. 
This can help in keeping members active as well as excited about being a part of your workgroup!

## Setting up a time

> When is good?

We have members and you have planned a nice agenda with consultation of the members -- great!

> The next step is to decide a time for the events such that you get the most participation and people also get a chance to digest things which they have attended. 
> The best thing you can do is ask the people about the timings which they are most comfortable with. 

Reviewer: ```suggestion
The next step is to decide a time for the events such that you get the most participation and people also get a chance to digest things which they have attended.

The best thing you can do is ask the people about the timings which they are most comfortable with. 
```

Consider making a poll for upcoming events with appropriate options for different timezones. 
If by any chance, the event gets cancelled or reschedule try to notify everyone about the update status as soon as possible to avoid any conflicts and save member's time. 
As workgroups evolve over time, consider reviewing your selected meeting schedule once or twice a year to see how members can best participate in the workgroup.

## What to do next with meetings

> To infinity and beyond!

You have now successfully organised an event and all members are happy with it too -- hooray! Here are some final thoughts as we conclude this blog post:

> In a meeting you won't have time to probably cover everything for everyone. Not all topics will be covered but instead what one can do is at the end of workgroup meetings, one can ask if someone wanted to discuss a topic or write down what topics were left over from the original agenda.

Reviewer: ```suggestion
In a meeting you won't have time to probably cover everything for everyone. Not all topics can be covered but instead what one can do is at the end of workgroup meetings, one can ask if someone wanted to discuss a topic or write down what topics were left over from the original agenda.
```
